app/controllers/document_images_controller.rb

```diff
@@ -0,0 +1,168 @@
+# frozen_string_literal: true
+
+require 'mini_magick'
+
+class DocumentImagesController < ApplicationController
+  def edit
+    @document = Document.find(params[:id])
+  end
+
+  def update
+    document = Document.find(params[:id])
+    upload = UploadedImageProcessorService.new(image_params[:image]).process
+
+    if !upload.valid?
+      redirect_to document_images_path(document),
+                   flash: { notice: "Image upload failed", errors: upload.errors }
+    else
+      blob = ActiveStorage::Blob.build_after_upload(
+        io: upload.file,
+        content_type: upload.mime_type,
+        filename: upload.filename,
+      )
+      blob.metadata[:width] = upload.dimensions[:width]
```

Comment:
Can we not give it a metadata hash when we build it?

Reply:
Nah, it doesn't accept one as an argument. I don't think we should use it for our metadata really since it feels like it could be semi-private to Blob, next round of this has us storing this information in our own model.